seed/views/v3/import_files.py

```diff
@@ -904,20 +904,21 @@ def greenbutton_meters_preview(self, request, pk):
                 {'status': 'error', 'message': 'Could not find import file with pk=' + str(
                     pk)}, status=status.HTTP_400_BAD_REQUEST)
 
-        parser = reader.GreenButtonParser(import_file.local_file)
-        raw_meter_data = list(parser.data)
-
         try:
             property_id = PropertyView.objects.get(pk=view_id, cycle__organization_id=org_id).property_id
         except PropertyView.DoesNotExist:
             return JsonResponse(
                 {'status': 'error', 'message': 'Could not find property with pk=' + str(
                     view_id)}, status=status.HTTP_400_BAD_REQUEST)
 
-        meters_parser = MetersParser(org_id, raw_meter_data, source_type=Meter.GREENBUTTON, property_id=property_id)
+        meters_parser = MetersParser.factory(
+            import_file.local_file,
+            org_id,
+            source_type=Meter.GREENBUTTON,
+            property_id=property_id
+        )
```

Comment:
lol not much of an improvement here, but I figured if I was creating the factory for both PM file types then I should probably includee greenbutton as well...